rust/datafusion/src/sql/utils.rs

```diff
@@ -335,3 +336,35 @@ where
         },
     }
 }
+
+/// Returns mapping of each alias (`String`) to the expression (`Expr`) it is
+/// aliasing.
+pub(crate) fn extract_aliases(exprs: &Vec<Expr>) -> HashMap<String, Expr> {
```

Comment:
This can probably be something like the following if you wanted more idomatic rust style (I don't think it really matters in this case)

```suggestion
pub(crate) fn extract_aliases(exprs: &[Expr]) -> HashMap<String, Expr> {
```

Maybe even something fancier like anything that can become an iterator:

```
pub(crate) fn extract_aliases(exprs: impl Iterator<Item=&Expr>) -> HashMap<String, Expr> {
```

Reply:
Thanks for the tip! I hadn't realized there's automatic deref coercion from vectors to slices. 

Updated in 3e52167.